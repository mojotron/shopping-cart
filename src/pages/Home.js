import React from "react";
import { Link } from "react-router-dom";

const Home = () => {
  return (
    <section>
      <h1>Home page</h1>
      <Link to="/shop">Shop</Link>
    </section>
  );
};

export default Home;
