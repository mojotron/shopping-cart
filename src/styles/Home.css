.Home {
  position: relative;
  margin: -10rem;
  height: 90vh;
  width: 100vw;
  background-image: url("../images/WallpaperDog-6356.jpg");
  background-color: black;
  background-size: cover;
  background-repeat: no-repeat;
}
.Home__headings {
  position: absolute;
  top: 20%;
  left: 30%;
  color: white;
  font-size: 4rem;
  text-transform: uppercase;
  text-shadow: 1px 1px 2px red, 0 0 1em blue, 0 0 0.2em blue;
  margin: 5rem;
}
.Home__headings > h2 {
  font-size: 2rem;
}
