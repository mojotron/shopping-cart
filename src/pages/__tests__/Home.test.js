import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../Home";

describe("Home page", () => {
  test("render Home snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  test("main heading with text content", () => {
    render(<Home />);
    expect(screen.getByText(/new fortnite goodies/i)).toBeInTheDocument();
  });

  test("h2 with text content", () => {
    render(<Home />);
    expect(screen.getByText(/explore the shop for/i)).toBeInTheDocument();
  });
});
