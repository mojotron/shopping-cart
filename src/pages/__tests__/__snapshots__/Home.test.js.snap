// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render Home snapshot 1`] = `
<div>
  <section
    class="Home"
  >
    <div
      class="Home__headings"
    >
      <h2>
        explore the shop for
      </h2>
      <h1>
        new fortnite goodies
      </h1>
    </div>
  </section>
</div>
`;
